src/averageAges.js

```diff
@@ -37,7 +42,17 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter((person) => person.sex === 'f');
+
+  const womenWithChildren = withChildren
+    ? women.filter(woman => people
+      .some(person => person.mother === woman.name))
+    : women;
+
+  const totalAge = womenWithChildren.reduce((sum, woman) => (
+    sum + woman.died - woman.born), 0);
+
+  return totalAge / womenWithChildren.length;
```

Comment:
you have the same logic in `men` function, please create a function helper for calculating average age